diff --git a/src/connection-editor/page-general.c b/src/connection-editor/page-general.c
index 794407d..96a2573 100644
--- a/src/connection-editor/page-general.c
+++ b/src/connection-editor/page-general.c
@@ -327,7 +327,7 @@ static void
 finish_setup (CEPageGeneral *self, gpointer user_data)
 {
 	CEPageGeneralPrivate *priv = CE_PAGE_GENERAL_GET_PRIVATE (self);
-	gboolean any_dependent_vpn;
+	gboolean any_dependent_vpn, vpn_visible;
 
 	priv->setup_finished = TRUE;
 
@@ -335,10 +335,15 @@ finish_setup (CEPageGeneral *self, gpointer user_data)
 
 	g_signal_connect (priv->firewall_zone, "changed", G_CALLBACK (stuff_changed), self);
 
+	vpn_visible = (gtk_combo_box_get_active (priv->dependent_vpn) != -1);
 	any_dependent_vpn = !!nm_setting_connection_get_num_secondaries (priv->setting);
-	gtk_toggle_button_set_active (priv->dependent_vpn_checkbox, any_dependent_vpn);
+	gtk_widget_set_visible (GTK_WIDGET (priv->dependent_vpn_checkbox), vpn_visible);
+	gtk_widget_set_visible (GTK_WIDGET (priv->dependent_vpn), vpn_visible);
+	if (vpn_visible) {
+		gtk_toggle_button_set_active (priv->dependent_vpn_checkbox, any_dependent_vpn);
+		gtk_widget_set_sensitive (GTK_WIDGET (priv->dependent_vpn), any_dependent_vpn);
+	}
 	g_signal_connect (priv->dependent_vpn_checkbox, "toggled", G_CALLBACK (vpn_checkbox_toggled), self);
-	gtk_widget_set_sensitive (GTK_WIDGET (priv->dependent_vpn), any_dependent_vpn);
 	g_signal_connect (priv->dependent_vpn, "changed", G_CALLBACK (stuff_changed), self);
 
 	g_signal_connect (priv->autoconnect, "toggled", G_CALLBACK (autoconnect_checkbox_toggled), self);
