diff --git a/src/connection-editor/page-wifi.c b/src/connection-editor/page-wifi.c
index a32f07a..48d01de 100644
--- a/src/connection-editor/page-wifi.c
+++ b/src/connection-editor/page-wifi.c
@@ -269,6 +269,7 @@ mode_combo_changed_cb (GtkComboBox *combo,
 	gboolean show_freq = TRUE;
 	gboolean show_bssid = TRUE;
 	gboolean hotspot = FALSE;
+	const char *band;
 
 	switch (gtk_combo_box_get_active (GTK_COMBO_BOX (combo))) {
 	case 1: /* hotspot */
@@ -306,6 +307,16 @@ mode_combo_changed_cb (GtkComboBox *combo,
 	gtk_widget_set_visible (GTK_WIDGET (priv->bssid), show_bssid);
 	gtk_widget_set_sensitive (GTK_WIDGET (priv->bssid), show_bssid);
 
+	band = nm_setting_wireless_get_band (priv->setting);
+	gtk_widget_set_sensitive (GTK_WIDGET (priv->channel), FALSE);
+	if (band) {
+		if (!strcmp (band, "a")) {
+			gtk_widget_set_sensitive (GTK_WIDGET (priv->channel), TRUE);
+		} else if (!strcmp (band, "bg")) {
+			gtk_widget_set_sensitive (GTK_WIDGET (priv->channel), TRUE);
+		}
+	}
+
 	ce_page_changed (CE_PAGE (self));
 }
 
