diff --git a/src/connection-editor/nm-connection-editor.c b/src/connection-editor/nm-connection-editor.c
index 25cc383..c384cc6 100644
--- a/src/connection-editor/nm-connection-editor.c
+++ b/src/connection-editor/nm-connection-editor.c
@@ -357,7 +357,8 @@ connection_editor_validate (NMConnectionEditor *editor)
 		if (!ce_page_validate (CE_PAGE (iter->data), editor->connection, &error)) {
 			if (!validation_error) {
 				validation_error = g_strdup_printf (_("Invalid setting %s: %s"),
-				                                    CE_PAGE (iter->data)->title,
+//				                                    CE_PAGE (iter->data)->title,
+                                                    "",
 				                                    error->message);
 			}
 			g_clear_error (&error);
